### ISSUE TITLE
Link with data-remote and data-confirm

### ISSUE THREAD

**Issue by Paxa** [hauleth/vanilla-ujs#14]
I have this link:
```html
<a data-method="delete" data-confirm="Delete airport TPE?" href="/admin/airports/TPE">Delete</a>
```

If I click 'Cancel' in confirm window, request is still being send.

Solution: move code to handle `'a[data-confirm], button[data-confirm]'` above code for `'a[data-method]'`. Problem happens because `data-method` handler create form and submit it, then `data-confirm` handler will try to stop event.

**Comment by walterdavis** [comment]
Sorry for the delay in looking at this. I don't think that changing the order of registration will help here, because the current confirm method only hooks the click event on links and buttons and inputs. We would also need to change the confirm handler to be triggered by an enclosing form's submit method. The method handler wraps the link in a form, then calls that form's submit method. I suspect that the reason the Rails UJS handler can deal with this is that it does not call submit immediately, but rather sends a synthetic event named something like `rails:submit`. This, in turn allows a last-ditch for confirmation, since it wraps around the actual form submission, whether that is a real form or one of these injected forms.

**Comment by walterdavis** [comment]
I tested this in a new stunt app, and found a different issue. If you add `method: :delete`, `remote: true` and `data: { confirm: 'Are you sure?' }` to a link, the cancel option works as expected. However, in a vanilla Rails 4.2.5 application, the controller attempts to redirect to a DELETE route, rather than a GET route (this is with a normal `resources :actors` method in my routes file).

~~~sh
Started DELETE "/actors/3" for 127.0.0.1 at 2016-03-05 11:22:12 -0500
Processing by ActorsController#destroy as */*
  Parameters: {"id"=>"3"}
  Actor Load (0.5ms)  SELECT  `actors`.* FROM `actors` WHERE `actors`.`id` = 3 LIMIT 1
   (0.2ms)  BEGIN
  SQL (0.6ms)  DELETE FROM `actors` WHERE `actors`.`id` = 3
   (3.4ms)  COMMIT
Redirected to http://localhost:3000/actors
Completed 302 Found in 8ms (ActiveRecord: 4.8ms)


Started DELETE "/actors" for 127.0.0.1 at 2016-03-05 11:22:12 -0500
```

When I removed all other format options from the destroy method in my actors_controller, it worked correctly, but what it looks like is we are not sending the correct request headers in the form for the controller to figure out that the request is a JS request. I will dig into this later if I get a chance.

**Comment by walterdavis** [comment]
Comparing these responses to a jQuery UJS delete link, I see that with jQuery, the server responds to the click as it should: 

```sh
Started DELETE "/things/19" for 127.0.0.1 at 2016-03-06 11:25:57 -0500
Processing by ThingsController#destroy as JS
  Parameters: {"id"=>"19"}
  Thing Load (0.5ms)  SELECT  "things".* FROM "things" WHERE "things"."id" = $1 LIMIT 1  [["id", 19]]
   (0.2ms)  BEGIN
  SQL (0.3ms)  DELETE FROM "things" WHERE "things"."id" = $1  [["id", 19]]
   (0.7ms)  COMMIT
   (0.7ms)  SELECT COUNT(*) FROM "things"
  Rendered things/destroy.js.erb (1.6ms)
Completed 200 OK in 14ms (Views: 9.0ms | ActiveRecord: 2.4ms)
```

Comparing the two in the Resources tab of Safari Web Inspector, I can see that jQuery's Ajax method is setting an Accepts: header with application/javascript and its variants, and it also sets a Content-Type: header with text/javascript. LiteAjax in vanilla-ujs is sending text/plain and no Accepts. I am going to see if this bug can be circumvented by doing either of these.

**Comment by walterdavis** [comment]
Okay, we have a winner. Adding a single (long) line to LiteJs gives us the desired result of having the remote: true links be noted "as JS" by the Rails controller. Adding another two lines ensures that even a tail-less URL will be treated as content-type 'text/javascript' rather than 'text/plain', although that is not really necessary to get the controller to wake up and consider the request to be a JS request. 

I am going to add a new issue and attach a patch request, since this came about in the wrong context.

**Comment by hauleth** [comment]
Should be fixed by #16. Thankx @walterdavis
